### PR TITLE
[#165547388] Fix issues with architect install and start

### PR DESCRIPTION
**Pull request by tjhiggins**

### CODE REVIEW

File: src/commands/start.ts [hunk 1]
```diff
@@ -150,6 +151,12 @@ export default class Start extends Command {
 
           resolve();
         });
+
+        cmd.on('error', (e) => {
+          this.log(_error(`Error: spawning architect-${service_config.language}-launcher`));
+          this.log(_error(e.toString()));
+          this.exit(1);
```
Comment (davidthor) [inline src/commands/start.ts]: We should bubble up this error in the promise rejection otherwise we have hanging promises.

**Comment by davidthor** [comment]
@tjhiggins looks pretty good and tests still pass on mac. Just be sure to file a follow-up ticket for the windows placeholder fix for the user IDs. If it works on windows as-is, then let's just remove that line from the docker run command on windows.

**Comment by tjhiggins** [comment]
https://www.pivotaltracker.com/story/show/165564921

File: src/commands/start.ts (hunk 1 above)
Comment (tjhiggins) [reply src/commands/start.ts]: So instead of calling exit:
`reject(error)`?

Comment (davidthor) [reply src/commands/start.ts]: Yup. We can catch errors at the top and log them.

**Pull request merged**